src/main/java/core/stack/MateStack.java

```diff
@@ -1,5 +1,50 @@
 package core.stack;
 
+import java.util.Objects;
+
 public class MateStack<T> {
+    private Node<T> firstElement;
+    private int iterator = 0;
+
+    public MateStack() {
+        this.firstElement = null;
+    }
+
+    public void push(T value) {
+        Node<T> firstPushElement = new Node<>();
+        if (firstPushElement == null) {
+            return;
+        }
+        firstPushElement.element = value;
+        firstPushElement.nextNode = firstElement;
```

Comment:
add and use constructor with all params to create new node

Reply:
@al-pavlenko done.